apps/testingaccessibility/src/context/progress-context.tsx

```diff
@@ -24,36 +26,37 @@ export function useProgress() {
 
 export const ProgressContext = React.createContext(defaultProgressContext)
 
-export const ProgressProvider: React.FC = ({children}) => {
-  const [isLoadingProgress, setIsLoadingProgress] =
-    React.useState<boolean>(true)
-  const [progress, setProgress] = React.useState<LessonProgress[]>([])
-  const {status} = useSession()
-
-  const fetchProgress = React.useCallback(async () => {
-    if (status === 'authenticated') {
-      const progress = await getLessonProgressForUser()
-      setProgress(progress)
-    }
+function useLoadProgress() {
+  const {status: sessionStatus} = useSession()
 
-    if (status !== 'loading') {
-      setIsLoadingProgress(false)
+  const {
+    data,
+    status: loadingProgressStatus,
+    refetch,
+  } = useQuery(['load progress', sessionStatus], async ({queryKey}) => {
+    if (last(queryKey) === 'authenticated') {
+      return await getLessonProgressForUser()
+    } else {
+      return []
     }
-  }, [status])
+  })
 
-  React.useEffect(() => {
-    if (status === 'authenticated') {
-      fetchProgress().catch(console.error)
-    }
-  }, [fetchProgress, status])
+  return {
+    progress: data,
+    status: loadingProgressStatus,
+    refetch,
+  }
+}
```

Comment:
`useQuery` does some nice caching and state management and works well for this type of loading activity

here I'm just wrapping it in a hook, but you could **probably** get away with just using the hook and not the context/provider.